core/iwasm/interpreter/wasm.h

```diff
@@ -626,11 +629,17 @@ wasm_get_smallest_type_idx(WASMType **types, uint32 type_count,
                            uint32 cur_type_idx)
 {
     uint32 i;
-
+#if WASM_ENABLE_TYPE_POINTER_COMPARE  != 0
+    for (i = 0; i < cur_type_idx; i++) {
+        if (types[cur_type_idx] != types[i])
```

Comment:
please inlight me here, L639 will return `i` when types are equal. But L634 will return `i` when types are not equal?